go/rpc/encrypted_client.go

```diff
@@ -122,7 +122,7 @@ func (c *EncRPCClient) CallContext(ctx context.Context, result interface{}, meth
 	return nil
 }
 
-func (c *EncRPCClient) Subscribe(ctx context.Context, namespace string, ch interface{}, args ...interface{}) (*rpc.ClientSubscription, error) {
+func (c *EncRPCClient) Subscribe(ctx context.Context, result interface{}, namespace string, ch interface{}, args ...interface{}) (*rpc.ClientSubscription, error) {
```

Comment:
If result is always a subscription ID could we have it be a return value with the Subscription? Or does this match a geth interface or something?

Reply:
I tried, but it doesn't come out very clean. The issue is that every client implements the same interface, so changing this forces `networkClient` to just return a nil value, which would then be overridden by enc rpc client based on the subscription contents.